crates/astria-sequencer/src/mempool/transactions_container.rs

```diff
@@ -159,6 +174,86 @@ impl PendingTransactionsForAccount {
     fn highest_nonce(&self) -> Option<u32> {
         self.txs.last_key_value().map(|(nonce, _)| *nonce)
     }
+
+    /// Removes and returns transactions that exceed the balances in `available_balances`.
+    fn find_demotables(
+        &mut self,
+        available_balances: &HashMap<IbcPrefixed, u128>,
+    ) -> Vec<TimemarkedTransaction> {
+        let mut split_at = 0;
+        let mut break_flag = false;
+        let mut available_balances = available_balances.clone();
+
+        for (nonce, tx) in &self.txs {
+            // ensure we have enough balance to cover inclusion
+            for (denom, cost) in tx.cost() {
+                match available_balances.entry(*denom) {
+                    hash_map::Entry::Occupied(mut entry) => {
+                        if cost <= entry.get() {
+                            // subtract cost
+                            let current_cost = entry.get_mut();
+                            *current_cost = current_cost.saturating_sub(*cost);
+                        } else {
+                            // not enough balance, do not include
+                            break_flag = true;
+                            break;
+                        }
+                    }
+                    hash_map::Entry::Vacant(_) => {
+                        // not enough balance, do not include
+                        if *cost != 0 {
+                            break_flag = true;
```

Comment:
i think you can use `break 'label` here as to not use this: https://doc.rust-lang.org/rust-by-example/flow_control/loop/nested.html